crates/ruma-common/src/api/metadata.rs

```diff
@@ -268,7 +268,7 @@ impl VersionHistory {
         if let Some(deprecated) = deprecated {
             if let Some(prev_seen_version) = prev_seen_version {
                 let ord_result = prev_seen_version.const_ord(&deprecated);
-                if ord_result.is_eq() {
+                if !deprecated.is_legacy() && ord_result.is_eq() {
```

Comment:
A comment here explaining what this is doing would be good

I know there is a comment on the public documentation, but here it'd be nice to restate it